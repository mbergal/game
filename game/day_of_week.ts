export type DayOfWeek =
    | "Monday"
    | "Tuesday"
    | "Wednesday"
    | "Thursday"
    | "Friday"
    | "Saturday"
    | "Sunday"

export const all: DayOfWeek[] = [
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
    "Sunday",
]
