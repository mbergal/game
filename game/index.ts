import { Command } from "../commands"
import { ItemGenerator } from "../objects/item"
import { GameMap } from "./map"
import * as Score from "./score"
export * as Score from "./score"

export type Game = {
    map: GameMap
    score: Score.Score
    itemGenerator: ItemGenerator
    commands: Command[]
}
