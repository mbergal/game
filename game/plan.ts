import * as Collapse from "./collapse"
import { Event } from "./event"
import * as PerformanceReview from "./performance_review"
import * as Sprint from "./sprint"

import _ from "lodash"
import config from "./config"

export type Plan = Map<number, Event.t[]>

export function make() {
    return new Map<number, Event.t[]>()
}

export function addEvent(plan: Plan, time: number, event: Event.t) {
    if (!plan.has(time)) {
        plan.set(time, [])
    }
    plan.get(time)!.push(event)
}

export function append(plan: Plan, other: Plan): Plan {
    for (const time of other.keys()) {
        for (const event of other.get(time)!) {
            addEvent(plan, time, event)
        }
    }
    return plan
}

export function generatePlan(startDay: number): Plan {
    let plan = make()
    let startTick = startDay * config.dayTicks
    for (const i in _.range(Math.floor((config.totalDays - startDay) / 14))) {
        const r = Sprint.generatePlan(startTick)
        append(plan, r[0])
        startTick = r[1]
    }
    startTick = startDay * config.dayTicks
    for (const i in _.range(Math.floor((config.totalDays - startDay) / 14))) {
        const pplan = PerformanceReview.generatePlan(startTick)
        append(plan, pplan)
        startTick += config.dayTicks * 14
    }

    append(plan, Collapse.plan())
    return plan
}
