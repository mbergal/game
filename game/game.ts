import { Command } from "../commands"
import * as ItemGenerator from "../game/item_generator"
import { Vector } from "../geometry"
import * as Player from "../objects/player"
import { assertUnreachable } from "../utils/utils"
import * as DayOfWeek from "./day_of_week"
import * as Effect from "./effect"
import * as Effects from "./effects"
import * as GameMap from "./map"
import { Message } from "./message"
import * as Plan from "./plan"
import * as Score from "./score"
import * as Sprint from "./sprint"
export * as GameMap from "./map"
export * as Score from "./score"

export type t = {
    map: GameMap.GameMap
    score: Score.Score
    time: GameTime
    itemGenerator: ItemGenerator.t
    sprint: Sprint.t
    commands: Command[]
    messages: Message[]
    messageTact: number
    player?: Player.Player
    plan: Plan.t
    messageStartTime: number | null
}

export type GameTime = {
    day: number
    ticks: number
    dayOfWeek: DayOfWeek.t
}

export type Game = t

export function make(size: Vector.t, plan: Plan.t): t {
    return {
        map: new GameMap.GameMap(size.x, size.y, []),
        commands: [],
        itemGenerator: ItemGenerator.make(),
        sprint: Sprint.make(),
        score: Score.make(),
        messages: [],
        messageTact: 0,
        time: {
            ticks: 0,
            day: 0,
            dayOfWeek: "Sunday",
        },
        plan: plan,
        messageStartTime: null,
    }
}

export function toJson(game: t): object {
    return {
        map: game.map.toJson(),
        score: game.score,
        itemGenerator: game.itemGenerator,
        commands: game.commands,
        messages: game.messages,
        messageTact: game.messageTact,
    }
}

export function handleEffects(game: t, effects: Generator<Effect.t> | Effects.t) {
    for (const effect of effects) {
        switch (effect.type) {
            case "null":
                break
            case "addImpact":
                game.score.impact += effect.impact
                break
            case "showMessage":
                message(game, effect.message)
                break
            default:
                assertUnreachable(effect)
        }
    }
}
export function message(game: t, m: Message | { text: string[]; ttl: number }) {
    if (m.text instanceof Array) {
        for (const text of m.text) {
            game.messages.push({ text: text, ttl: m.ttl })
        }
    } else {
        game.messages.push({ text: m.text, ttl: m.ttl })
    }
}

export interface GameStorage {
    save(json: string): void
    load(): string | null
}

export function save(game: Game, storage: GameStorage) {
    storage.save(JSON.stringify(toJson(game)))
    console.log("Game saved!")
}

export function load(storage: GameStorage): Game | null {
    const objectsStorage = storage.load()
    if (objectsStorage != null) {
        const {
            commands,
            itemGenerator,
            score,
            sprint,
            messages,
            messageTact,
            map,
            time,
            plan,
            messageStartTime,
        }: {
            score: Score.Score
            sprint: Sprint.t
            commands: Command[]
            messages: Message[]
            messageTact: number
            itemGenerator: ItemGenerator.t
            map: object
            time: GameTime
            plan: Plan.t
            messageStartTime: number
        } = JSON.parse(objectsStorage)
        const map_ = GameMap.GameMap.fromJson(map)
        const player = map_.objects.find<Player.t>((x): x is Player.t => x.type === "player")
        return {
            time: time,
            score: score,
            itemGenerator: itemGenerator,
            sprint: sprint,
            messages: messages,
            messageTact: messageTact,
            commands: commands,
            map: map_,
            player: player,
            plan: plan,
            messageStartTime: messageStartTime,
        }
    } else {
        console.log("There is no saved game.")
        return null
    }
}
