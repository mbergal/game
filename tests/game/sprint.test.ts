import { expect, test } from "vitest"

import _ from "lodash"
import * as Plan from "../../game/plan"

test("generatePlan", () => {
    const plan = Plan.generatePlan(0)
    const a = _.chain(plan)
        .entries()
        .map(([key, value]) => [key.toString().padStart(4), value] as const)
        .sortBy((x) => parseInt(x[0]))
        .fromPairs()
        .value()
    expect(a).toMatchSnapshot()
})
