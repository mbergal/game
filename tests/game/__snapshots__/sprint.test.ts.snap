// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`a 1`] = `
{
  "   0": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": 0,
      "type": "createBacklogIssue",
    },
  ],
  "  17": [
    {
      "size": 0,
      "type": "createBacklogIssue",
    },
  ],
  "  33": [
    {
      "size": 0,
      "type": "createBacklogIssue",
    },
  ],
  "  50": [
    {
      "size": 1,
      "type": "createBacklogIssue",
    },
  ],
  "  67": [
    {
      "size": 1,
      "type": "createBacklogIssue",
    },
  ],
  "  83": [
    {
      "size": 2,
      "type": "createBacklogIssue",
    },
  ],
  " 100": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  " 182": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  " 199": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  " 200": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  " 299": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  " 300": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  " 399": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  " 400": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  " 499": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  " 500": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  " 701": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  " 800": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  " 801": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  " 900": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  " 901": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "1000": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "1001": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "1100": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "1101": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "1302": [
    {
      "type": "weekendEnd",
    },
  ],
}
`;

exports[`generatePlan 1`] = `
{
  "   0": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "  17": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "  33": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "  50": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "  67": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "  83": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  " 182": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  " 183": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  " 282": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  " 283": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  " 382": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  " 383": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  " 482": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  " 483": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  " 582": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  " 583": [
    {
      "type": "weekendStart",
    },
  ],
  " 784": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  " 883": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  " 884": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  " 983": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  " 984": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "10077": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "10078": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "10177": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "10178": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "10277": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "10278": [
    {
      "type": "weekendStart",
    },
  ],
  "10479": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "10578": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "10579": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "10678": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "10679": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "10778": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "10779": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "1083": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "1084": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "10878": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "10879": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "11080": [
    {
      "type": "weekendEnd",
    },
  ],
  "1183": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "1184": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "1385": [
    {
      "type": "weekendEnd",
    },
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "1402": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "1418": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "1435": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "1452": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "1468": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "1567": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "1568": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "1667": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "1668": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  "175895": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "175912": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "175928": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "175945": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "175962": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "175978": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "176077": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "176078": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "176177": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "176178": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  "176277": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "176278": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "176377": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "176378": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "176477": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "176478": [
    {
      "type": "weekendStart",
    },
  ],
  "176679": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "1767": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "176778": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "176779": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "1768": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "176878": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "176879": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "176978": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "176979": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "177078": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "177079": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "177280": [
    {
      "type": "weekendEnd",
    },
  ],
  "1867": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "1868": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "1967": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "1968": [
    {
      "type": "weekendStart",
    },
  ],
  "20775": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "20792": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "20808": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "20825": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "20842": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "20858": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "20957": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "20958": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "21057": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "21058": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  "21157": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "21158": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "21257": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "21258": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "21357": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "21358": [
    {
      "type": "weekendStart",
    },
  ],
  "21559": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "21658": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "21659": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "2169": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "21758": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "21759": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "21858": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "21859": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "21958": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "21959": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "22160": [
    {
      "type": "weekendEnd",
    },
  ],
  "2268": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "2269": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "2368": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "2369": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "2468": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "2469": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "2568": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "2569": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "2770": [
    {
      "type": "weekendEnd",
    },
  ],
  "353175": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "353192": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "353208": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "353225": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "353242": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "353258": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "353357": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "353358": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "353457": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "353458": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  "353557": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "353558": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "353657": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "353658": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "353757": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "353758": [
    {
      "type": "weekendStart",
    },
  ],
  "353959": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "354058": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "354059": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "354158": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "354159": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "354258": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "354259": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "354358": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "354359": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "354560": [
    {
      "type": "weekendEnd",
    },
  ],
  "4155": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "4172": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "4188": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "4205": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "4222": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "4238": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "42935": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "42952": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "42968": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "42985": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "43002": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "43018": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "43117": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "43118": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "43217": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "43218": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  "43317": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "43318": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "4337": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "4338": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "43417": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "43418": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "43517": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "43518": [
    {
      "type": "weekendStart",
    },
  ],
  "43719": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "43818": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "43819": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "43918": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "43919": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "44018": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "44019": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "44118": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "44119": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "44320": [
    {
      "type": "weekendEnd",
    },
  ],
  "4437": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "4438": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  "4537": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "4538": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "4637": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "4638": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "4737": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "4738": [
    {
      "type": "weekendStart",
    },
  ],
  "4939": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "5038": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "5039": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "5138": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "5139": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "5238": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "5239": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "5338": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "5339": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "5540": [
    {
      "type": "weekendEnd",
    },
  ],
  "707735": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "707752": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "707768": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "707785": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "707802": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "707818": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "707917": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "707918": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "708017": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "708018": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  "708117": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "708118": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "708217": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "708218": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "708317": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "708318": [
    {
      "type": "weekendStart",
    },
  ],
  "708519": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "708618": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "708619": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "708718": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "708719": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "708818": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "708819": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "708918": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "708919": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "709120": [
    {
      "type": "weekendEnd",
    },
  ],
  "87255": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "87272": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "87288": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "87305": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "87322": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "87338": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "87437": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "87438": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "87537": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "87538": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
  "87637": [
    {
      "day": 2,
      "type": "sprintDayEnd",
    },
  ],
  "87638": [
    {
      "day": 3,
      "type": "sprintDayStart",
    },
  ],
  "87737": [
    {
      "day": 3,
      "type": "sprintDayEnd",
    },
  ],
  "87738": [
    {
      "day": 4,
      "type": "sprintDayStart",
    },
  ],
  "87837": [
    {
      "day": 4,
      "type": "sprintDayEnd",
    },
  ],
  "87838": [
    {
      "type": "weekendStart",
    },
  ],
  "88039": [
    {
      "type": "weekendEnd",
    },
    {
      "day": 5,
      "type": "sprintDayStart",
    },
  ],
  "88138": [
    {
      "day": 5,
      "type": "sprintDayEnd",
    },
  ],
  "88139": [
    {
      "day": 6,
      "type": "sprintDayStart",
    },
  ],
  "88238": [
    {
      "day": 6,
      "type": "sprintDayEnd",
    },
  ],
  "88239": [
    {
      "day": 7,
      "type": "sprintDayStart",
    },
  ],
  "88338": [
    {
      "day": 7,
      "type": "sprintDayEnd",
    },
  ],
  "88339": [
    {
      "day": 8,
      "type": "sprintDayStart",
    },
  ],
  "88438": [
    {
      "day": 8,
      "type": "sprintDayEnd",
    },
  ],
  "88439": [
    {
      "type": "sprintEnd",
    },
    {
      "type": "weekendStart",
    },
  ],
  "88640": [
    {
      "type": "weekendEnd",
    },
  ],
  "9695": [
    {
      "type": "sprintStart",
    },
    {
      "type": "groomBacklogStart",
    },
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "9712": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "9728": [
    {
      "size": "small",
      "type": "createBacklogIssue",
    },
  ],
  "9745": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "9762": [
    {
      "size": "medium",
      "type": "createBacklogIssue",
    },
  ],
  "9778": [
    {
      "size": "large",
      "type": "createBacklogIssue",
    },
  ],
  "9877": [
    {
      "type": "groomBacklogEnd",
    },
  ],
  "9878": [
    {
      "day": 1,
      "type": "sprintDayStart",
    },
  ],
  "9977": [
    {
      "day": 1,
      "type": "sprintDayEnd",
    },
  ],
  "9978": [
    {
      "day": 2,
      "type": "sprintDayStart",
    },
  ],
}
`;
